### PR TITLE
Fix speech command dropdown and update commands in ET example

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview

Updates the keywords and labels in the ET sample to no longer clash with [system keywords](https://docs.microsoft.com/en-us/hololens/hololens-cortana#general-speech-commands). The new keywords are "record eye gaze" and "end recording".

image

While investigating this, I also noticed some general speech command issues that I fixed up, including

1. The KeywordAndResponse not filling the entire inspector

    image

2. Voice command not being settable from a Unity event

    image

I unified the speech command dropdown behavior into a single utility class, which is now used across `Interactable` and `SpeechInputHandler`, which previously had their own distinct implementations.

Additionally, `SpeechInputHandler` had a bug from porting from HTK that was preventing the custom dropdown from showing up in the first place.

## Changes
- Fixes https://github.com/microsoft/MixedRealityToolkit-Unity/issues/9363, fixes https://github.com/microsoft/MixedRealityToolkit-Unity/issues/8122, and fixes https://github.com/microsoft/MixedRealityToolkit-Unity/issues/10181

### CODE REVIEW

File: Assets/MRTK/Core/Inspectors/Utilities/SpeechKeywordUtility.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        public static void RenderKeywordsExcept(string[] keywordsInUse, SerializedProperty property, Rect rect = default(Rect), GUIContent content = null)
+        {
+            IEnumerable<string> availableKeywords = GetDistinctRegisteredKeywords();
+
+            if (keywordsInUse != null && availableKeywords != null)
+            {
+                availableKeywords = availableKeywords.Except(keywordsInUse.Select(keyword => keyword)).ToArray();
+            }
+
+            bool validSpeechKeywords = availableKeywords != null;
```
Comment (RogPodge) [inline Assets/MRTK/Core/Inspectors/Utilities/SpeechKeywordUtility.cs:35]: This should also check if availableKeywords is empty
```suggestion
            bool validSpeechKeywords = availableKeywords != null && availableKeywords.Count() > 0;
```

**Review by RogPodge — CHANGES_REQUESTED** [review]

File: Assets/MRTK/Core/Inspectors/Utilities/SpeechKeywordUtility.cs (hunk 1 above)
Comment (RogPodge) [reply Assets/MRTK/Core/Inspectors/Utilities/SpeechKeywordUtility.cs:35]: Looks good otherwise!

Comment (keveleigh) [reply Assets/MRTK/Core/Inspectors/Utilities/SpeechKeywordUtility.cs:35]: Ooh good catch!

**Review by RogPodge — APPROVED** [review]

**Pull request merged**